include/swift/AST/DiagnosticsDriver.def

```diff
@@ -88,7 +88,7 @@ WARNING(warning_unnecessary_repl_mode,none,
         "unnecessary option '%0'; this is the default for '%1' "
         "with no input files", (StringRef, StringRef))
 ERROR(error_unsupported_option,none,
-      "unsupported option '%0' for '%1'; did you mean '%2 %0'?",
+      "the option '%0' is not supported by '%1'; did you mean to use '%2'?",
```

Comment:
We tend to [keep our diagnostics terse](https://github.com/apple/swift/blob/master/docs/Diagnostics.md#grammar-and-phrasing), partly so that it's more likely to fit on one line of output or in an IDE. In this case I think your rephrasing is mostly good, but I would drop the initial "the".

(The `error_i_mode` diagnostic above is fairly old and not one that comes up very much, since that change was made a long time ago.)